src/components/implementation/no_interface/cFE_booter/osfilesys.c

```diff
@@ -0,0 +1,326 @@
+    file -> name        = location;
+    if (*(location+strlen(location) - 1) == '/') {
+        file -> type    = FSOBJ_DIR;
+        file -> size    = 0;
+    }
+    else{
+        file -> type    = FSOBJ_FILE;
+        file -> size    = oct_to_dec(location + 124);
+        file -> data    = location + 500;
+    }
+    return OS_FS_SUCCESS;
+}
+
+//this could be more performate bithack
+uint32 round_up(uint32 offset)
+{
+    if (offset % 512) return offset + (512 - (offset % 512));
+    return offset;
+}
+
+/******************************************************************************
+** fsobj Level Methods
+******************************************************************************/
+
+uint32 fsobj_init(struct fsobj *o)
+{
+    assert(o);
+    assert(o->filedes > 0);
+
+    o->name     = "EXAMPLE_NAME";
+	o->type     = FSOBJ_DIR;
+	o->size     = 0;
+    o->refcnt   = 0;
+	o->data     = NULL;
+	o->child    = o->parent = NULL;
+    o->next     = o->prev = NULL;
+    return OS_FS_SUCCESS;
+}
+
+//finds the next open file
+uint32 newfile_get(struct fsobj **o)
+{
+    uint32 count = 0;
+    while (count < MAX_NUM_FILES && files[count].filedes) {
+        count++;
+    }
+    if (count == MAX_NUM_FILES) return OS_FS_ERROR;
+    *o =  &files[count];
+    (*o) -> filedes=count+1;
+    //filedes needs to be unique and nonzero, so filedes is defined as index+1
+    return OS_SUCCESS;
+}
+
+uint32 file_insert(struct fsobj *o, struct FS *filesys)
+{
+    assert(filesys && filesys->root && filesys->root->filedes);
+    if (filesys->root == o) return OS_SUCCESS;
+
+    insert(o, filesys->root);
+
+    return OS_SUCCESS;
+}
+
+//could be written iteratively to use less stackframe if that becomes an issue
+// Root is
+uint32 insert(struct fsobj *o, struct fsobj *root)
+{
+    assert(root && o);
+    assert(strlen(root->name) < strlen(o->name));
+
+    //loop terminates when it finds a place to put o, either at the end of a
+    //list of children or as the first in an empty children list
+    while (1) {
+        assert(strcmp(root->name, o->name) !=0 );
+        assert(strspn(o->name, root->name) >= strlen(root->name));
+        //if there is no child, then insert as child
+        if (!root->child) {
+            o->parent = root;
+            root->child = o;
+            return OS_SUCCESS;
+        }
+        root = root->child;
+
+        //precondition: root is the first in a non-empty list of children
+        //postcondition: root is an ancestor of o
+        while (strspn(o->name, root->name) < strlen(root->name)) {
+            if (!root->next) {
+                root->next = o;
+                o->prev = root;
+                o->parent = root->parent;
+                return OS_SUCCESS;
+            }
+            root = root->next;
+        }
+
+    }
+    assert(0);
+    return OS_FS_ERROR;
+}
+
+int32 open(const char *path) {
+    assert(fs[0].root && !fs[1].root);
+    struct fsobj *o = fs[0].root;
+    o = find(path, o);
```

Comment:
I find the reuse of the `o` variable confusing here. I'd suggest splitting it into to `root` and `o`.